lib/Platform/Intl/java/com/facebook/hermes/intl/DateTimeFormat.java

```diff
@@ -379,25 +390,8 @@ private void initializeDateTimeFormat(List<String> locales, Map<String, Object>
     }
   }
 
-  private String normalizeTimeZoneName(String timeZoneName) {
-    // https://tc39.es/ecma402/#sec-case-sensitivity-and-case-mapping
-    // Note that we should convert only upper case translation in ASCII range.
-    StringBuilder normalized = new StringBuilder(timeZoneName.length());
-    int offset = 'a' - 'A';
-    for (int idx = 0; idx < timeZoneName.length(); idx++) {
-      char c = timeZoneName.charAt(idx);
-      if (c >= 'a' && c <= 'z') {
-        normalized.append((char) (c - offset));
-      } else {
-        normalized.append(c);
-      }
-    }
-
-    return normalized.toString();
-  }
-
   private boolean isValidTimeZoneName(String timeZone) {
-    return mPlatformDateTimeFormatter.isValidTimeZone(timeZone);
+    return Pattern.compile("^[\\x00-\\x7F]+[a-zA-Z]+$").matcher(timeZone).find();
   }
```

Comment:
We should be able to eliminate this function altogether now.

Reply:
Interesting, in the case of comparing timezones, it doesn't explicitly forbid non-ASCII characters, it just says that we have to convert to uppercase before comparing, and when when we compare to uppercase, we should only convert lowercase ASCII characters. That's what the current implementation (before this PR) of `normalizeTimeZoneName` does. So let's follow exactly what the spec says here. Instead of doing validation with a regex, we should convert both the argument and the timezone we are comparing it to to uppercase (only for ASCII characters), and then compare those to determine if there is a match.

Does that sound reasonable?

Reply:
Instead, should we just write our own string comparator which only operates on a-z, as described in https://tc39.es/ecma402/#sec-case-sensitivity-and-case-mapping? That seems simpler than having a separate step involving an ad hoc regexp check.